src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -234,6 +235,26 @@ public bool TryLoadConfig(
 
                 config = RuntimeConfig;
 
+                // When isDevMode is not null it means we are in a hot-reload scenario, and need to save the previous
+                // mode in the new RuntimeConfig since we do not support hot-reload of the mode.
+                if (isDevMode is not null && config.Runtime is not null && config.Runtime.Host is not null)
+                {
+                    // Log error when the mode is changed during hot-reload. 
+                    if (isDevMode.Value && HostMode.Production.Equals(config.Runtime.Host.Mode))
+                    {
```

Comment:
what if mode is attempted to be changed from PRoduction to Development? No warning logged then?

Reply:
That shouldnt be possible for now, but since we will have production file monitoring in the future, can make this check cover that case as well for when we add in the functionality.